src/Type/Php/ArrayKeyExistsFunctionTypeSpecifyingExtension.php

```diff
@@ -57,13 +58,37 @@ public function specifyTypes(
 		$array = $node->getArgs()[1]->value;
 		$keyType = $scope->getType($key);
 		$arrayType = $scope->getType($array);
-		if (!$keyType instanceof ConstantIntegerType && !$keyType instanceof ConstantStringType && !$arrayType->isIterableAtLeastOnce()->no()) {
+
+		if (!$keyType instanceof ConstantIntegerType
+			&& !$keyType instanceof ConstantStringType
+			&& !$arrayType->isIterableAtLeastOnce()->no()) {
 			if ($context->truthy()) {
+				$arrayKeyType = $arrayType->getIterableKeyType();
+				if ((new StringType())->isSuperTypeOf($keyType)->yes()) {
+					$arrayKeyType = $arrayKeyType->toString();
+				}
+
+				$specifiedTypes = $this->typeSpecifier->create(
+					$key,
+					$arrayKeyType,
+					$context,
+					false,
+					$scope,
+				);
+
 				$arrayDimFetch = new ArrayDimFetch(
 					$array,
 					$key,
 				);
-				return $this->typeSpecifier->create($arrayDimFetch, $arrayType->getIterableValueType(), $context, false, $scope, new Identical($arrayDimFetch, new ConstFetch(new Name('__PHPSTAN_FAUX_CONSTANT'))));
+
+				return $specifiedTypes->unionWith($this->typeSpecifier->create(
+					$arrayDimFetch,
+					$arrayType->getIterableValueType(),
+					$context,
+					false,
+					$scope,
+					new Identical($arrayDimFetch, new ConstFetch(new Name('__PHPSTAN_FAUX_CONSTANT')))
```

Comment:
```suggestion
					new Identical($arrayDimFetch, new ConstFetch(new Name('__PHPSTAN_FAUX_CONSTANT'))),
```
missing trailing comma as also reported by phpcs

Reply:
Fixed